browser/brave_ads/search_result_ad/search_result_ad_tab_helper.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_ads/search_result_ad/search_result_ad_tab_helper.h"
+
+#include "base/feature_list.h"
+#include "brave/browser/brave_ads/ads_service_factory.h"
+#include "brave/components/brave_ads/browser/ads_service.h"
+#include "brave/components/brave_ads/common/features.h"
+#include "brave/components/brave_ads/content/browser/search_result_ad/search_result_ad_handler.h"
+#include "brave/components/brave_search/common/brave_search_utils.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/render_frame_host.h"
+
+namespace brave_ads {
+
+namespace {
+
+AdsService* g_ads_service_for_testing = nullptr;
+
```

Comment:
Nit: remove empty lines, Chromium appears to do this for single entries

Reply:
Fixed